WMCO 4.0.2 release notes (#57272)
openshift/openshift-docs: windows_containers/windows-containers-release-notes-4-x.adoc

If you do not have this additional Red Hat subscription, you can use the Community Windows Machine Config Operator, a distribution that lacks official support. 
endif::openshift-origin[]

[id="wmco-4-0-2"]
== Release notes for Red Hat Windows Machine Config Operator 4.0.2

Issued: 2023-4-17

> The WMCO 4.0.2 is now available with bug fixes. The components of the WMCO were released in link:https://access.redhat.com/errata/.

Author: @opayne1  Sorry. I should have noted that I don't have the Live ID yet. I will put that in later. (TBH, I just want to get this one as far along as possible to get it off my plate!)

[id="wmco-4-0-2-bug-fixes"]
=== Bug fixes

> * Previously, the containers on the Windows nodes were assigned a hard-coded DNS server IP address (such as 172.30.0.10) that was based on the default subnet. This caused DNS resolution to fail in certain situations. This fix removes the hard-coded DNS address. Windows containers are now assigned a valid DNS server IP address that is based on the cluster DNS configuration. As a result, DNS resolution works as expected for Windows containers. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2020350[**BZ#2020350**])

Reviewer: Just a minimalism suggestion to remove some of the extra articles. Up to you if you would like to take them. 
```suggestion
* Previously, containers on Windows nodes were assigned a hard-coded DNS server IP address (such as 172.30.0.10) based on the default subnet. This caused DNS resolution to fail in certain situations. This fix removes the hard-coded DNS address, and Windows containers are now assigned a valid DNS server IP address based on the cluster DNS configuration. As a result, DNS resolution works as expected for Windows containers. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2020350[**BZ#2020350**])
```

Reviewer: Also, I wasn't sure about "[**BZ#2020350**])" should this be two * or should it be "[*BZ#2020350*])"

I see in 4.0.1 it is listed with two so disregard if it is supposed to be that way.

> * Windows Containers are assigned a hardcoded DNS server address of 172.30.0.10, if the default subnet is not used, the Windows containers on Windows nodes were assigned a wrong DNS Server IP. As a result, DNS resolution does not work. With this fix, if the default subnet is not used, the hard-coded DNS information is removed and Windows containers on Windows nodes get assigned a valid DNS server IP address. DNS resolution works as expected for Windows workloads. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2020350[**BZ#2020350**])

Reviewer 2: > As a result, DNS resolution does not work. With this fix, if the default subnet is not used, the hard-coded DNS information is removed and Windows containers on Windows nodes get assigned a valid DNS server IP address.

Consider rewording as:
```
This fix removes the hard-coded DNS information and the containers on the
Windows nodes get assigned a valid DNS server IP address based on the cluster
DNS configuration. 
```
